eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessor.java

```diff
@@ -69,26 +80,47 @@ public SubSnpNoHgvs process(SubSnpNoHgvs subSnpNoHgvs) throws Exception {
             processedContigs.add(contigName);
         }
 
-        if (contigPresentInAssemblyReport) {
+        if (isChromosomeReplaceable(subSnpNoHgvs, chromosomeSynonyms)) {
+            replaceChromosomeWithGenbankAccession(subSnpNoHgvs, chromosomeSynonyms);
+        } else if (isContigReplaceable(contigSynonyms)) {
             replaceContigWithGenbankAccession(subSnpNoHgvs, contigSynonyms);
+        } else {
+            // No replacement is possible. We must keep the original RefSeq accession
         }
 
         return subSnpNoHgvs;
     }
 
-    private void replaceContigWithGenbankAccession(SubSnpNoHgvs subSnpNoHgvs, ContigSynonyms contigSynonyms) {
-        if ((contigSynonyms.isIdenticalGenBankAndRefSeq() || isGenbank(assemblyAccession)) &&
-                contigSynonyms.getGenBank() != null) {
-            subSnpNoHgvs.setContigName(contigSynonyms.getGenBank());
-        } else {
-            // genbank is not identical to refseq and the assembly is not genbank, so
-            // we must keep the original refseq
+    private boolean isChromosomeReplaceable(SubSnpNoHgvs subSnpNoHgvs, ContigSynonyms chromosomeSynonyms) {
+        StringBuilder reason = new StringBuilder();
+
+        boolean replaceable;
+        try {
+            replaceable = ContigSynonymValidationProcessor.isChromosomeReplaceable(
+                    subSnpNoHgvs.getChromosome(),
+                    subSnpNoHgvs.getChromosomeStart() != null,
+                    chromosomeSynonyms,
+                    reason);
+        } catch (IllegalStateException e) {
```

Comment:
Could we either create a new exception that makes clearer what is going on, or document this better? If an exception happens, the least I would expect is to continue with the replacement (but I understood the logic after reading the implementation for method `ContigSynonymValidationProcessor.isChromosomeReplaceable`).